### PR TITLE
Multi-tasking support

### LINKED ISSUE #708: Editor does not work with iPad multitasking in iOS9
The editor is belly up when used within iPad multitasking in iOS 9:

editor - make bigger

editor - slide in

editor - split

/cc @thianhlu

### PR DESCRIPTION
**Pull request by bummytime**
Fixes #708 

This PR adds iOS multitasking support to the editor:

scaled screen shot 2015-09-15 at 6 24 47 pm

scaled screen shot 2015-09-16 at 11 44 08 am

scaled screen shot 2015-09-15 at 6 25 39 pm

scaled screen shot 2015-09-15 at 6 25 05 pm

**Things to note in this PR:**

* The format bar is now created within a xib (`WPEditorFormatbarView.xib`) and uses size classes and auto layout (no more layouts in code). There are two distinct size class layouts: `Compact—Any` and `Regular—Any`
* The `WPEditorFormatbarView` class owns everything within in it (e.g. no more passing in an array of format bar items from `WPEditorViewController`), therefore the `WPEditorFormatbarViewDelegate` has been updated accordingly.
* The iPad and iPhone format bar heights are now both 44pts —this helps support regular and compact size classes better (e.g. landscape on an iPhone 6+)
* The HTML mode button is now a `< >` icon instead of the text `HTML`
* The vertical border separating the HTML button on smaller screens (compact size) has been removed
* The left & right editor margins were reduced on iPad to 40pts

**Things to test:**

1. Test all multitasking modes (on iPad 2 Air device if possible) — slide over, split view, primary app, secondary app, etc
2. The editor should work on iOS9 and on all devices
3. All format bar animations should work
4. All format bar buttons should work. 
5. Is the post preserved during multitasking changes? (e.g. do we lose content)
6. **RTL Language Support:** multitasking will be reversed

Note: There is a rotation issue that will be handled outside of this PR — will follow up in a comment on this PR.

/cc @astralbodies thanks for taking a look at this!

### CODE REVIEW

**Comment by bummytime** [comment]
@astralbodies I logged the rotation issue in #736. If you find anything related to it feel free to append comments/findings there.

**Comment by bummytime** [comment]
Just as an update to a convo @astralbodies and I had offline, in testing on device he saw the format bar resize issue using the following steps (no rotation needed):

1. Kill the app
2. Launch full screen with keyboard showing.
3. Home button out to main screen.
4. Launch Notes.
5. Swipe out 1/3 floating screen of EditorDemo - leave keyboard up (edited)
6. Dock 1/3.
7. Swipe out 1/2 screen. Keyboard should still be up, toolbar shows half compressed.

Not able to repro on the sim. :-( Luckily I finally got my hands on a iPad Pro to check this out for myself.

**Comment by astralbodies** [comment]
Video of the bug: https://cloudup.com/cXkEpVW2TgB

**Comment by astralbodies** [comment]
The resizing of that bar was fixed with that commit!

One issue I'm seeing as I'm mid-drag of the 1/3 floating window:

**Comment by bummytime** [comment]
@astralbodies that issue appears to be a function of our input acc view swizzling. Would you be ok with me logging that as a separate issue outside of this PR?

**Comment by astralbodies** [comment]
@bummytime Totes :smile:

**Comment by bummytime** [comment]
@astralbodies ok logged that issue in #740. Let me know if there is anything else you find.

File: Classes/UIView+SizeClass.m [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+- (BOOL)isViewHorizontallyCompact
+{
+    // iOS <= 8:
+    // We'll just consider 'Compact' all of non iPad Devices
+    if ([self respondsToSelector:@selector(traitCollection)] == false) {
+        return ([UIDevice currentDevice].userInterfaceIdiom == UIUserInterfaceIdiomPad) == false;
+    }
+    
+    return self.traitCollection.horizontalSizeClass == UIUserInterfaceSizeClassCompact;
```
Comment (astralbodies) [inline Classes/UIView+SizeClass.m]: You could dump iOS 8 support :smile:

File: Classes/WPEditorFormatbarView.m [hunk 2]
```diff
@@ -0,0 +1,528 @@
...
+
+- (void)initBarButtonItem:(ZSSBarButtonItem*)barButtonItem
+                  withTag:(WPEditorViewControllerElementTag)tag
+             htmlProperty:(NSString*)htmlProperty
+                imageName:(NSString*)imageName
+                   target:(id)target
+                 selector:(SEL)selector
+       accessibilityLabel:(NSString*)accessibilityLabel
+{
+    NSAssert(barButtonItem, @"ZSSBarButtonItem should not be nil here.");
```
Comment (astralbodies) [inline Classes/WPEditorFormatbarView.m]: I like using `NSParameterAssert` but I think that's just personal preference not a rule. :smile:

**Comment by astralbodies** [comment]
I say :shipit: :grinning:

File: Classes/UIView+SizeClass.m (hunk 1 above)
Comment (bummytime) [inline Classes/UIView+SizeClass.m]: :tada:

**Comment by bummytime** [comment]
Thanks @astralbodies !

**Pull request merged**